### PR TITLE
📖 docs: update CAPI support and guarantees

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

Update support and guarantees

cc @kubernetes-sigs/cluster-api-release-team

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
/lgtm
/hold

Until the release date passes so we don't have to update :sweat_smile:

**Comment by sbueringer** [comment]
/lgtm
/approve

Not sure I understand why we want to hold. I assume so we don't have to update in case we can't make the release date?
(but of course I'm fine with holding :))

**Review by furkatgofurov7 — COMMENTED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -152,14 +152,15 @@ Cluster API maintains the most recent release/releases for all supported API and
...
-| v1.3.x        | **v1beta1**  | when v1.5.0 will be released                         |
-| v1.2.x        | **v1beta1**  | when v1.4.0 will be released, tentatively March 2023 |
-| v1.1.x        | **v1beta1**  | EOL since 2022-07-18 - v1.2.0 release date (*)       |
-| v1.0.x        | **v1beta1**  | EOL since 2022-02-02 - v1.1.0 release date (*)       |
-| v0.4.x        | **v1alpha4** | EOL since 2022-04-06 - API version EOL               |
-| v0.3.x        | **v1alpha3** | EOL since 2022-02-23 - API version EOL               |
+| Minor Release | API Version  | Supported Until                                |
+|---------------|--------------|------------------------------------------------|
+| v1.4.x        | **v1beta1**  | when v1.6.0 will be released                   |
+| v1.3.x        | **v1beta1**  | when v1.5.0 will be released                   |
```
Comment (fabriziopandini) [inline CONTRIBUTING.md:158]:
```suggestion
| v1.3.x        | **v1beta1**  | when v1.5.0 will be released, tentatively July 2023                |
```

**Review by fabriziopandini — COMMENTED** [review]
Thanks for preparing this PR
Also, what about adding EOL on all the mentions to 1.2 in https://cluster-api.sigs.k8s.io/reference/versions.html; ideally we should also drop 1.1 columns and add 1.5 columns, but I'll defer to you if to do this in one PR or to split the work

File: CONTRIBUTING.md (hunk 1 above)
Comment (oscr) [reply CONTRIBUTING.md]: No problem and we got plenty of time. Added and updated.

**Pull request merged**